libs/runtime/ebpf_epoch.c

```diff
@@ -817,18 +871,20 @@ _ebpf_epoch_messenger_commit_release_epoch(
     // Set the released_epoch to the value computed by the EBPF_EPOCH_CPU_MESSAGE_TYPE_PROPOSE_RELEASE_EPOCH message.
     cpu_entry->released_epoch = message->message.commit_epoch.released_epoch - 1;
 
+    next_cpu = _ebpf_epoch_next_active_cpu(current_cpu);
+
     // If this is the last CPU, send the message to the first CPU to complete the cycle.
-    if (current_cpu != _ebpf_epoch_cpu_count - 1) {
-        // Send the message to the next CPU.
-        next_cpu = current_cpu + 1;
-    } else {
+    if (next_cpu == 0) {
         message->message_type = EBPF_EPOCH_CPU_MESSAGE_TYPE_PROPOSE_EPOCH_COMPLETE;
-        next_cpu = 0;
     }
 
     _ebpf_epoch_send_message_async(message, next_cpu);
 
     _ebpf_epoch_release_free_list(cpu_entry, cpu_entry->released_epoch);
```

Comment:
If a CPU goes active between the propose and commit phase, everything on the free list will be marked with a value lower than the released epoch.

If we made the sentinel UINT64_MAX (and made epoch uint64_t everywhere) then it wouldn't free anything until it knows the current epoch.

Reply:
Unknown epoch has to be low to prevent other CPUs from releasing memory that this CPU might be using, hence using 0 for unknown epoch, but you are correct about the problem if we activate between propose and commit. We should only release if current epoch not equal to the sentinel value.